src/main.rs

```diff
@@ -164,13 +166,21 @@ fn main() -> anyhow::Result<()> {
     };
 
     let toolchains = match &criterion {
```

Comment:
This type is getting pretty complicated.
```suggestion
    // `None`: do not remove based on toolchain version
    // `Some(None)`: remove all installed toolchains
    // `Some(Some(Vec))`: remove only the specified toolchains
    let toolchains = match &criterion {
```